### DISCUSSION TITLE
How to smoothly migrate away from /auth

### DISCUSSION THREAD

**Discussion by pschichtel** [keycloak/keycloak#22554] [Archive]
I've just upgraded my keycloak from 18-legacy all the way to 22. As part of this I've configured keycloak with `--http-relative-path /auth` so my existing clients don't break.

This keycloak instance has a bunch of realms, each with their own frontend URL (which also includes the `/auth` prefix).

Now I want to migrate from `/auth` to just `/`, but for that I need some why to have a smooth transition, since there are quite a few clients that need to be reconfigured. I can't change them all at once.

My original plan was to simply remove the `--http-relative-path /auth` option and introduce a redirect from `^/auth(?:/(.*))?$` to `/$1` in the nginx in front of keycloak. That seemed to work at first, but any clients attempting authentication failed with "Invalid request" in keycloak.

Has anyone else done this before?

**Comment by danielFesenmeyer** [comment]
Are you sure its ANY clients that can't authenticate? Or just clients using existing tokens or trying to refresh tokens?

The authentication failure probably occurs because Keycloak puts the realm url as issuer claim into tokens - and validates it for each authentication. When you configure Keycloak without `--http-relativepath /auth`, the _/auth_ path is no longer part of the issuer. When a client created a token with the old `/auth` configuration, the issuers won't match and the token is considered invalid.

The mentioned check is done here: https://github.com/keycloak/keycloak/blob/8299f151ede434a04b08b4e4a5a11b79d7b23871/services/src/main/java/org/keycloak/services/managers/AuthenticationManager.java#L1435
(Not completely sure whether it's the only location)
That behavior is not configurable or changeable via SPI, so I guess there is no simple solution to solve that issue probably.

**Reply by pschichtel** [reply]
I don't think this is the issue here, because it doesn't even get to the point of creating a token. And it happens for new sessions: The application redirects to keycloak for the authentication flow but keycloak just displays "Invalid request" after login.
